src/sagemaker/jumpstart/estimator.py

```diff
@@ -729,11 +730,27 @@ def attach(
 
         model_version = model_version or "*"
 
+        additional_kwargs = {"model_id": model_id, "model_version": model_version}
+
+        model_specs = verify_model_region_and_return_specs(
+            model_id=model_id,
+            version=model_version,
+            region=sagemaker_session.boto_region_name,
+            scope=JumpStartScriptScope.TRAINING,
+            tolerate_deprecated_model=True,  # model is already trained, so tolerate if deprecated
+            tolerate_vulnerable_model=True,  # model is already trained, so tolerate if vulnerable
```

Comment:
question: if a model is vulnerable, would this code raise an exception alerting the user:
```py
my_estimator = estimator.attach()
my_model = my_estimator.deploy()
```

Reply:
Nevertheless, it seems that you are only retrieving the specs to check if the model is gated or not, so that is non-blocking. Consider updating the comments to provide more details however.

Reply:
If the model was actually vulnerable or deprecated, you would get an exception when doing `my_estimator.deploy()`. As you mention, these tolerate flags are just to get the status of whether the model is gated or not